src/main/scala/firrtl/transforms/Flatten.scala

```diff
@@ -36,55 +43,47 @@ class Flatten extends InlineInstances {
        }
      }
 
-   // Replicate the circuit such that we inline only the modules that are below hierarchy given by the annotations
-   // Along with replication we rewrite the original circuit to refer to the new modules that will be inlined
+   /**
+    *  Modifies the circuit by replicating the hierarchy under the annotated objects (mods and insts) and
+    *  by rewriting the original circuit to refer to the new modules that will be inlined later
+    */
    def duplicateSubCircuitsFromAnno(c : Circuit, mods:  Set[ModuleName], insts : Set[ComponentName], modsToInline : mutable.Set[ModuleName]) : Circuit = {
      val modMap = c.modules.map(m => m.name->m) toMap
      val seedMods = mutable.Map.empty[String, String]
      val newModDefs= mutable.Set.empty[DefModule]
-     
-     // We start with rewriting original modules that contain annotations
+     val nsp = Namespace(c)
+
+     /** We start with rewriting original modules that contain annotations.  Populates seedMods */
      def rewriteMod(parent : DefModule)(x : Statement) : Statement = x match {
```

Comment:
All type signatures should have the `:` immediately after the name, then a space, then the type, e.g.:
```scala
def rewriteMod(parent: DefModule)(x: Statement): Statement = x match {
```
Please fix all instances of this.